app/controllers/api/base_controller.rb

```diff
@@ -1,3 +1,26 @@
 class Api::BaseController < ApplicationController
   layout nil
-end
+
+  def authenticate_user
+    email = params[:email]
+    password = params[:password]
+
+    user = User.find_by(email: email)
+
+    if user.nil? || !user.password_secure_compare(password)
+      render json: { error: 'Invalid credentials' }, status: 401
```

Comment:
Using `password_secure_compare` directly on the user object might expose sensitive logic. Consider using a service or a method within the User model to handle authentication securely.